harvester/utils/general_utils.py

```diff
@@ -721,3 +722,24 @@ def is_valid_uuid4(uuid_string) -> bool:
         AttributeError
     ):  # Catches cases where input might not be a string (e.g., UUID(0))
         return False
+
+
+class UnsafeTemplateEnvError(RuntimeError):
+    pass
+
+
+def render_block(template_name: str, block_name: str, **context) -> Response:
+    """
+    Render a specific block from a Jinja template, while using the Flask's default environment.
+    """
+    env = current_app.jinja_env
+    if not getattr(env, "autoescape", None):
+        raise UnsafeTemplateEnvError(
+            "Jinja autoescape is disabled; enable it or use Flask's jinja_env."
+        )
+    template = env.get_template(template_name)
+
+    # Render only the named block (Jinja will still escape vars inside the block)
+    block_gen = template.blocks[block_name]
+    html = "".join(block_gen(template.new_context(context)))
+    return Response(html, mimetype="text/html; charset=utf-8")
```

Comment:
This all looks fine, but I'm not sure this is the right location. The harvester folder is mostly for the harvest runner, the tasks that get created to process a job.
I would put this somewhere in the app context, either in init, routes, or some other location.